SandWorm/MeshCompareComponent.cs

```diff
@@ -0,0 +1,174 @@
+﻿using System;
+using System.Collections.Generic;
+using Rhino.Geometry.Intersect;
+using Grasshopper.Kernel;
+using Rhino.Geometry;
+
+namespace SandWorm
+{
+    public class CompareMeshes
+    {
+        public CompareMeshes()
+        {
+        }
+
+        private double[] _meshElevationPoints;
+
+        public double[] MeshElevationPoints
+        {
+            get { return _meshElevationPoints; }
+            set { _meshElevationPoints = value; }
+        }
+    }
+    public class MeshCompareComponent : GH_Component
+    {
+        public List<GeometryBase> outputSurface;
+        private Curve inputRectangle;
+        private double scaleFactor = 1;
+        private double sensorElevation;
+        public int leftColumns = 0;
+        public int rightColumns = 0;
+        public int topRows = 0;
+        public int bottomRows = 0;
+        public SetupOptions options; // List of options coming from the SWSetup component
+        public CompareMeshes results; // Array of resulting elevation points 
+        public List<string> output;
+        public Mesh inputMesh;
+
+        public MeshCompareComponent()
+          : base("MeshCompare", "MeshCompare",
+              "Visualizes elevation differences between meshes.",
+              "Sandworm", "Sandbox")
+        {
+        }
+
+        /// <summary>
+        /// Registers all the input parameters for this component.
+        /// </summary>
+        protected override void RegisterInputParams(GH_Component.GH_InputParamManager pManager)
+        {
+            pManager.AddCurveParameter("Rectangle", "RC", "Rectangle", GH_ParamAccess.item);
+            pManager.AddMeshParameter("Mesh", "M", "Mesh to be compared to the Kinect's scan", GH_ParamAccess.item);
+            pManager.AddNumberParameter("Scale Factor. 1 : ", "SF", "Scale factor for the referenced terrain.", GH_ParamAccess.item, scaleFactor);
```

Comment:
I think the help text here should explain more clearly what the purpose of the parameter is in terms of what the user might want to do with it. I think the use of the `1 : ` is also a little confusing (e.g. I wouldn't think that it is interpreted as a simple float multiplier and could be set as `0.5` or `1 : 0.5`. 

Also, going on precedent with the native GH components, perhaps it might be better labelled as "Scaling Factor"? Its a subtle change, but I think breaking associations with paper scale terminology makes its use more clear.

Reply:
Yeah, I'm not really happy with the way how this is handled. I'm running a SandBox class with some first years as I type it and they are having a hard time understanding it too :)